azure-quantum/azure/quantum/optimization/streaming_problem.py

```diff
@@ -0,0 +1,322 @@
+    :type terms: Optional[List[Term]], optional
+    :param init_config: Optional configuration details, depending on solver. Defaults to None
+    :type init_config: Optional[Dict[str,int]], optional
+    :param problem_type: Problem type (ProblemType.pubo or ProblemType.ising), defaults to ProblemType.ising
+    :type problem_type: ProblemType, optional
+    """
+    def __init__(
+        self,
+        workspace: Workspace,
+        name: str = 'Optimization Problem',
+        terms: Optional[List[Term]] = None,
+        init_config: Optional[Dict[str,int]] = None,
+        problem_type: ProblemType = ProblemType.ising,
+        metadata: Dict[str, str] = {},
+        **kw
+    ):
+        super(StreamingProblem, self).__init__(**kw)
+        self.name = name
+        self.id = str(uuid.uuid1())
+        self.workspace = workspace
+        self.problem_type = problem_type
+        self.init_config = init_config
+        self.terms_queue = Queue()
+        self.uploaded_uri = None
+        self.upload_to_url = None
+        self.uploader = None
+        self.compress = True
+        self.__n_couplers = 0
+        self.stats = {
+            'type': problem_type.name,
+            'max_coupling': 0,
+            'avg_coupling': 0,
+            'min_coupling': sys.maxsize,
+            'num_terms': 0,
+        }
+        self.upload_size_threshold = 10e6
+        self.upload_terms_threshold = 1000
+        self.metadata = metadata
+        if terms is not None and len(terms) > 0:
+            self.add_terms(terms.copy())
+
+    def add_term(self, c: Union[int, float], indices: List[int]) : 
+        """Adds a single term to the `Problem` representation
+
+        :param c: The cost or weight of this term
+        :type c: int, float
+        :param indices: The variable indices that are in this term
+        :type indices: List[int]
+        """
+        self.add_terms([Term(indices=indices, c=c)])
+    
+    def _get_upload_coords(self):
+        blob_name=self.id
+        if self.upload_to_url:
+            blob_client = BlobClient.from_blob_url(self.upload_to_url)
+            container_client = ContainerClient.from_container_url(
+                self.workspace._get_linked_storage_sas_uri(
+                    blob_client.container_name))
+            blob_name = blob_client.blob_name
+        elif not self.workspace.storage:
+            # No storage account is passed, use the linked one
+            container_uri = self.workspace._get_linked_storage_sas_uri(self.id)
+            container_client = ContainerClient.from_container_url(container_uri)
+        else:
+            # Use the specified storage account
+            container_client = ContainerClient.from_connection_string(self.workspace.storage, self.id)
+        
+        return { 'blob_name': blob_name, 'container_client': container_client }
+
+    def add_terms(self, terms: List[Term]) :
+        """Adds a list of terms to the `Problem` representation
```

Comment:
Would you add a note here about what's happening behind the scenes with uploading the terms?